django_irods/storage.py

```diff
@@ -372,7 +372,27 @@ def size(self, name):
         if "CAT_NO_ROWS_FOUND" in stdout:
             raise ValidationError("{} cannot be found in iRODS to retrieve "
                                   "file size".format(name))
-        return int(stdout)
+        return int(float(stdout))
+
+    def checksum(self, name):
+        # first force checksum (re)computation
+        self.session.run("ichksum", None, "-f", name)
+        # retrieve checksum using iquest
+        if '/' in name:
```

Comment:
It's not clear to me why this is being done and I'm not sure if it belongs in this method.  Maybe the name parameter should be cleaned before being passed to this method?

Reply:
The name parameter is the data object path, which in iRODS term, is the collection plus data object name. The logic here is just to check whether the name parameter include collection path by checking if slash is in the name parameter. I thought this is self-explanatory, but apparently not. I can add more comment for clarification or put it in an internal function to extract object name from a path.